java8/src/main/java/Client/LogicalOperations.java

```diff
@@ -0,0 +1,184 @@
+package Client;
+
+import Data.Car;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class LogicalOperations {
+
+    static List<Car> getObjects(Car t1, Car t2, Car t3, Car t4, Car t5, Car t6) {
+
+        List<Car> list = List.of(t1, t2, t3, t4, t5, t6);
+        return list;
+    }
+
+    static HashSet<Car> filterCarsWithTriggers(List<Car> carslist) {
+        HashSet<Car> targetSet = new HashSet<>();
+        List<Car> intermediaryList = carslist.stream()
+                .filter(car -> car.getModel().startsWith("M"))
+                .collect(Collectors.toList());
+
+        for (Car car : intermediaryList) {
+            String model = car.getModel();
+            char[] modelToChar = model.toCharArray();
+            int count = 0;
+
+            for (char ch : modelToChar)
+                if (ch == 'a') {
+                    count++;
+                    if (count > 1) {
+
+                        System.out.println(modelToChar);
+                        targetSet.add(car);
+                    }
+                }
+        }
+        return targetSet;
+    }
+
+    static void addRandomString(HashSet<String> set) {
+
+        int firstLetter = 97;
+        int lastLetter = 122;
+        int boundary = generateRandomInt();
+        Random random = new Random();
+
+        String generated = random.ints(firstLetter, lastLetter)
+                .limit(boundary)
+                .collect(StringBuilder::new, StringBuilder::appendCodePoint, StringBuilder::append)
+                .toString();
+
+        System.out.println(generated);
+        set.add(generated);
+    }
+
+    /**
+     * When calling .max with .valueOf(char c), the  first chars from the char[] composing the given strings
+     * are compared using "natural order",
+     *
+     * @return the string with the biggest first element.
+     * << public static String valueOf(char[] data) {
+     * return new String(data); >> }
+     */
+    static void findMax(HashSet<String> set) {
+        String max = String.valueOf(
+                set.stream()
+                        .max(Comparator.comparing(s -> s.valueOf(s))));
+        System.out.println(max);
+    }
+
+    static int countElementsOver10(Map<String, Integer> map) {
+
+        int count = 0;
+        for (Map.Entry<String, Integer> entry : map.entrySet()) {
+            Integer v = entry.getValue();
+            if (v > 10) count++;
+        }
+        return count;
+    }
+
+    static void printIfEligible(List<Car> carsList) {
+        carsList.stream().filter(LogicalOperations::isEligible).map(car -> car.getModel().toUpperCase() + " has 4 doors.").forEach(System.out::println);
+    }
+
+    private static boolean isEligible(Car car) {
+        if (car.getDoors() != 4) {

```

Comment:
I was expecting the actual usage of .match(), easier to implement, try it, but your conditions work as well (try simplifying them)